pkg/auth/auth.go

```diff
@@ -43,6 +44,38 @@ func CheckAuthFile(authfile string) error {
 	return nil
 }
 
+// Mirrors --authfile to a tmpfile if --authfile points to a
+// file descriptor instead of actual file on filesystem
+// reason is file descriptor could vanish mid-session if consumed
+// once, check https://github.com/containers/buildah/issues/3070
+func CreateTempAuthFileIfNeeded(authfile string) string {
+	if strings.HasPrefix(authfile, "/dev/fd") {
+		b, err := ioutil.ReadFile(authfile)
+		if err != nil {
+			// if anything goes wrong return authfile
+			// and let c/common/auth handle failues later
+			return authfile
+		}
+		tmpfile, err := ioutil.TempFile(os.TempDir(), "temp-common-auth")
+		if _, err := tmpfile.Write(b); err != nil {
+			// if anything goes wrong return authfile
+			// and let c/common/auth handle failues later
+			return authfile
+		}
+
+		return filepath.Join(os.TempDir(), tmpfile.Name())
```

Comment:
This looks wrong,  doesn't tmpFile.NAME() include the path?  Won't this create something like

/var/tmp/var/tmp/TMPFILE

Reply:
@rhatdan you are correct. fixed in latest commit.